### DISCUSSION TITLE
Java Auto Instrumentation - HTTP Transaction Name

### DISCUSSION THREAD

**Discussion by baole229801** [open-telemetry/opentelemetry-java-instrumentation#11994] [Q&A]
My application has a very long URL Path for API, for example:
/services/data/v1/SmartOTPObject/operations/SmartOTPActivation/OmniSmartOTPAllocate

When it appears in Kibana, everything is grouped as /services/data/v1/*
How can I disable this grouping? Thank you.

**Comment by laurit** [comment]
> How can I disable this grouping? Thank you.

This can't be disabled with configuration. Have a look at https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name Specification requires that a low cardinality name be used. To get a low cardinality name we have instrumentations for a variety of frameworks. Without knowing anything about your application it is not possible to tell why we are not able to use the full path as a low cardinality name. It could be that we don't have an instrumentation for the framework you use. If you wish for someone to take a look please provide a minimal application that reproduces the issue.
It is possible to manually instrument your application to provide better span names.

**Reply by baole229801** [reply]
Thank you for your help @laurit
